src/components/TradeWidget/OrderValidity.tsx

```diff
@@ -282,14 +283,49 @@ const OrderValidity: React.FC<Props> = ({
     validUntilInputId,
     validUntilInputValue,
   ])
+  const validFromRef: React.MutableRefObject<HTMLInputElement | null> = useRef(null)
+  const validUntilRef: React.MutableRefObject<HTMLInputElement | null> = useRef(null)
+
+  // This side effect is for not requiring disable on validFrom/Until inputs
+  // and auto-magically updating the checkbox/values on change
+  // also allows auto focus and select when manually unchecking ASAP or Never checkboxes
+  useEffect(() => {
+    // undefined validFrom input - set ASAP
+    !validFromInputValue
+      ? batchedUpdates(() => {
+          setAsap(true)
+          setValue(validFromInputId, undefined, true)
+        })
+      : setAsap(false)
+    // undefined validUntil input - set unlimited
+    !validUntilInputValue
+      ? batchedUpdates(() => {
+          setUnlimited(true)
+          setValue(validUntilInputId, undefined, true)
+        })
+      : setUnlimited(false)
+  }, [setAsap, setUnlimited, setValue, validFromInputValue, validFromInputId, validUntilInputValue, validUntilInputId])
 
   function handleUnlimitedClick(): void {
+    const reffedInput = validUntilRef.current!
     setUnlimited(isUnlimited => !isUnlimited)
-    !isUnlimited ? setValue(validUntilInputId, undefined, true) : setValue(validUntilInputId, '2880', true)
+    if (!isUnlimited) {
+      return setValue(validUntilInputId, undefined, true)
+    }
+
+    reffedInput.focus()
+    setValue(validUntilInputId, '2880', true)
+    reffedInput.select()
   }
   function handleASAPClick(): void {
+    const reffedInput = validFromRef.current!
     setAsap(isAsap => !isAsap)
-    !isAsap ? setValue(validFromInputId, undefined, true) : setValue(validFromInputId, '30', true)
+    if (!isAsap) {
+      return setValue(validFromInputId, undefined, true)
+    }
+    reffedInput.focus()
```

Comment:
I'm ok with this. Anyways, I'm curious. The other method we used for the "sell amount" didn't work?